File: PyProject/leetcode/two-sum.py
from typing import List


class TwoSum:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = dict()
        for i, v in enumerate(nums):
            complementary = target - v
            if m.get(complementary) is not None:
                return [m.get(complementary), i]
            m[v] = i

File: GoProject/leetcode/array/two-sum.go
package array

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)

	l := len(nums)

	for i := 0; i < l; i++ {
		t := target - nums[i]

		v, ok := m[t]
		if ok {
			return []int{v, i}
		} else {
			m[nums[i]] = i
		}
	}
	return nil

}
